### PR TITLE
Empty HandlerBox name instead of "libavif"

### PR DESCRIPTION
**Pull request by y-guyon**
Empty HandlerBox name by default.

### CODE REVIEW

File: include/avif/avif.h [hunk 1]
```diff
@@ -216,6 +216,8 @@ typedef enum avifHeaderFormat
 {
     // AVIF file with an "avif" brand, a MetaBox and all its required boxes for maximum compatibility.
     AVIF_HEADER_FULL,
+    // Same as AVIF_HEADER_FULL but the name field of the HandlerBox is empty instead of "libavif".
```
[the remark this answers is not in the document]
Comment (wantehchang) [reply include/avif/avif.h:219]: @leo-barnes Leo: This is the libavif  HandlerBox name issue I mentioned at the end of the AOMedia Image Container subgroup meeting today.

Yannis: I suggest we make this change without adding a new AVIF header format option.

[the remark this answers is not in the document]
Comment (leo-barnes) [reply include/avif/avif.h:219]: I can confirm that Apple platforms create HEIF files with empty handler name.

[the remark this answers is not in the document]
Comment (y-guyon) [reply include/avif/avif.h]: > Yannis: I suggest we make this change without adding a new AVIF header format option.

Done. PTAL.

File: src/write.c [hunk 2]
```diff
@@ -771,6 +771,18 @@ static avifResult avifEncoderWriteExtendedColorProperties(avifRWStream * dedupSt
     return AVIF_RESULT_OK;
 }
 
+static avifResult avifEncoderWriteHandlerBox(avifRWStream * s, const char handler_type[4])
```
Comment (wantehchang) [inline src/write.c:774]: Nit: The function name should start with `avifRWStream` (the type of the first/main parameter) intead of `avifEncoder`.

Comment (wantehchang) [inline src/write.c:774]: Nit: handler_type => handlerType

File: tests/gtest/avifheadertest.cc [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  ImagePtr image = testutil::CreateImage(12, 34, 8, AVIF_PIXEL_FORMAT_YUV420,
+                                         AVIF_PLANES_ALL);
+  ASSERT_NE(image, nullptr);
+  testutil::FillImageGradient(image.get());
+  const testutil::AvifRwData avif = testutil::Encode(image.get());
+  ASSERT_NE(avif.data, nullptr);
+
+  // Make sure the HandlerBox is as small as possible, meaning its name field is
+  // empty.
+  const uint8_t* kHdlr = reinterpret_cast<const uint8_t*>("hdlr");
```
Comment (wantehchang) [inline tests/gtest/avifheadertest.cc:23]: Nit: The way `kHdlr` is defined here, it is a variable, not a constant. (It is not even a const variable.) So it should be named like a variable:

```
    const uint8_t* const hdlr = reinterpret_cast<const uint8_t*>("hdlr");
```

Alternatively, we can define it as a constant:

```
    const uint8_t kHdlr[] = { 'h', 'd', 'l', 'r' };
```

File: tests/gtest/avifheadertest.cc [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  uint8_t* hdlr_position =
```
Comment (wantehchang) [inline tests/gtest/avifheadertest.cc:24]: Nit: add `const`:

```
  const uint8_t* hdlr_position =
```

**Review by wantehchang — APPROVED** [review]
LGTM. Thanks. I suggest some changes.

**Pull request merged**